cgk_calcium_tools/filehandler.py

```diff
@@ -120,116 +104,183 @@ def __init__(
-
-        if recording_labels is None:
-            self.recording_labels = self.rec_paths
-        else:
-            assert len(recording_labels) == len(
-                self.rec_paths
-            ), "Recordings and reconding labels should have same length"
-            self.recording_labels = recording_labels
-
-        # Lookig for multiplanes:
-        self.p_rec_paths = []
-        self.p_outputsfolders = []
-        self.p_recording_labels = []
-        self.focus_files = {}
-        self.efocus = []
-        for i, rec in enumerate(self.rec_paths):
-            raw_gpio_file = os.path.splitext(rec)[0] + ".gpio"  # raw data for gpio
-            updated_gpio_file = (
-                os.path.splitext(rec)[0] + "_gpio.isxd"
-            )  # after the first reading gpio is converted to this
-            local_updated_gpio_file = os.path.join(
-                self.outputsfolders[i], Path(updated_gpio_file).name
-            )  # new gpio copied in output
-            if os.path.exists(local_updated_gpio_file):
-                efocus = get_efocus(local_updated_gpio_file)
-            elif os.path.exists(updated_gpio_file):
-                efocus = get_efocus(updated_gpio_file)
-            elif os.path.exists(raw_gpio_file):
-                local_raw_gpio_file = os.path.join(
-                    self.outputsfolders[i], Path(raw_gpio_file).name
-                )
-                shutil.copy2(raw_gpio_file, local_raw_gpio_file)
-                efocus = get_efocus(local_raw_gpio_file)
-            else:
-                video = isx.Movie.read(rec)
-                get_acquisition_info = video.get_acquisition_info().copy()
-                del video  # usefull for windows
-                if "Microscope Focus" in get_acquisition_info:
-                    assert isx.verify_deinterleave(
-                        rec, get_acquisition_info["Microscope Focus"]
-                    ), f"Info {rec}: Multiple Microscope Focus but not gpio file"
-                    efocus = [get_acquisition_info["Microscope Focus"]]
+                if one_file_per_folder:
+                    assert len(files) == 1, "Multiple files found for {}.".format(
+                        str(Path(mainf) / subfolder / fpatter)
+                    )
                 else:
-                    efocus = [0]
-                    print(f"Info: Unable to verify Microscope Focus config in: {rec}")
-            if len(efocus) == 1:
-                self.focus_files[rec] = [rec]
-                self.p_rec_paths.append(rec)
-                self.p_recording_labels.append(self.recording_labels[i])
-                self.efocus.append(efocus)
+                    files = [r for r in files if r not in meta["rec_paths"]]
+                for file in files:
+                    if not overwrite_metadata:
+                        json_file = os.path.splitext(file)[0] + "_metadata.json"
+                        if os.path.exists(json_file):
+                            continue
```

Comment:
if exist already and not overwrite_metadata, never loads the meta dict form the json. Then is not available for processing